workers/unity/Assets/Gdk/Core/GameObjectRepresentation/SpatialOSGameObjectCreator.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    /// <summary>
+    ///     Creates a companion gameobject for newly spawned entities according to a prefab definition.
+    /// </summary>
+    public class SpatialOSGameObjectCreator
+    {
+        private readonly World world;
+        private readonly Dictionary<string, GameObject> cachedPrefabs;
+
+        public SpatialOSGameObjectCreator(World world, Dictionary<string, GameObject> initialCachedPrefabs)
```

Comment:
Is there a reason this takes a dictionary but during the usage we pass it an empty dictionary? 

(I assume this is for the future configuration for this, but wanted to sanity check)

Reply:
This is to make unit tests work. In the unit test, we give the constructor a non-empty dictionary to avoid having to actually look for a prefab on disc.